### ISSUE TITLE
Join forces with zfit?

### ISSUE THREAD

**Issue by HDembinski** [mramospe/minkit#12]
Hi, I just stumbled over your project. Nice to see iminuit used. Judging from the example, it looks like you are trying to make a RooFit clone. Perhaps you could join forces with https://github.com/zfit/zfit? They are also working on such a thing.

**Comment by mramospe** [comment]
Hi Hans, this package was born before I had knowledge about zfit. I created it to do fast fits in python using C++ and/or CUDA. The idea was to make a ROOT independent framework for fitting, purely python based (the similarities with RooFit are inevitable). I haven't used zfit quite a lot, since I haven't needed to do fits after I finished my previous analysis. I simply tested a couple of things and it looks really nice.

I am up to join the effort, since I always thought we lack a strong pythonic package to do fits to probability density functions, and zfit is by far more evolved than minkit.

**Comment by HDembinski** [comment]
I am not personally involved in zfit, but I think it would be great to have more good people on the project. Maybe you can contact Jonas Eschle or just drop by their repo with an issue.

If you are a good programmer, starting from scratch is of course much faster than joining a bigger project, where you have to argue about every change :). But in the long run, having these larger projects which have the critical mass to really attract a lot of users is the key.

**Comment by HDembinski** [comment]
Just one more note: I played around with numba and it really generates fantastically fast code, it can also use cuda. I think that's the way to go in the long run, not C++.

**Comment by mramospe** [comment]
Indeed, things have evolved quite a lot in the GPU side, and relying in some external package to handle the C/CUDA interface is a better approach. When I started doing fits in GPUs I only had PyCUDA to deal with CUDA from python, but now there are other more smart alternatives. I haven't done speed tests comparing both, though.
